controllers/discovery/searchDomain.js

```diff
@@ -11,7 +11,7 @@ const {sequelize} = require('../../databases/models');
 const SearchDomain = async (req, res) => {
   const {q} = req.query;
   const {userId} = req;
-  if (q.length < 2)
+  if (q?.length < 2 || !q)
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The condition `!q` should be checked before `q?.length < 2`. If `q` is `null` or `undefined`, the current order will still throw an error.
```diff
- if (q?.length < 2 || !q)
+ if (!q || q.length < 2)
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
implement in the new commit